Runtime/Hosting/NativeHost.cs

```diff
@@ -0,0 +1,172 @@
+
+internal partial class NativeHost : IDisposable
+{
+    private const string ManagedHostAssemblyName = nameof(NodeApi);
+    private const string ManagedHostTypeName =
+        $"{nameof(NodeApi)}.{nameof(NodeApi.Hosting)}.ManagedHost";
+
+    private hostfxr_handle _hostContextHandle;
+
+    [UnmanagedCallersOnly(
+        EntryPoint = nameof(napi_register_module_v1),
+        CallConvs = new[] { typeof(CallConvCdecl) })]
+    public static napi_value InitializeModule(napi_env env, napi_value exports)
+    {
+        ////Console.WriteLine("> NativeHost.InitializeModule()");
+
+        try
+        {
+            using var scope = new JSValueScope(env);
+
+            // Constructing the native host also loads the CLR and initializes the managed host.
+            var host = new NativeHost(env, exports);
+
+            // The managed host defined several properties/methods already.
+            // Add on a dispose method implemented by the native host that closes the CLR context.
+            new JSValue(scope, exports).DefineProperties(new JSPropertyDescriptor(
+                nameof(NativeHost.Dispose),
+                (_) => { host.Dispose(); return default; }));
+        }
+        catch (Exception ex)
+        {
+            Console.Error.WriteLine($"Failed to load CLR native host module: {ex}");
+        }
+
+        ////Console.WriteLine("< NativeHost.InitializeModule()");
+
+        return exports;
+    }
+
+    public NativeHost(napi_env env, napi_value exports)
+    {
+        string currentModuleFilePath = GetCurrentModuleFilePath();
+        ////Console.WriteLine("Current module path: " + currentModuleFilePath);
+        string nodeApiHostDir = Path.GetDirectoryName(currentModuleFilePath)!;
+
+        string runtimeConfigJsonPath = Path.Join(nodeApiHostDir, @"NodeApi.runtimeconfig.json");
+        ////Console.WriteLine("CLR config: " + runtimeConfigJsonPath);
+
+        string managedHostAsssemblyPath = Path.Join(nodeApiHostDir, @"NodeApi.dll");
+        ////Console.WriteLine("Managed host: " + managedHostAsssemblyPath);
+
+        // Load the library that provides CLR hosting APIs.
+        HostFxr.Initialize();
+
+        // https://github.com/vmoroz/napi-cs/blob/dev/NodeApi.Sdk.CLR/Build/nativehost.cpp
+
+        // Initialize the CLR with configuration from runtimeconfig.json.
+        hostfxr_status status = hostfxr_initialize_for_runtime_config(
+            runtimeConfigJsonPath, initializeParameters: default, out _hostContextHandle);
+        CheckStatus(status, "Failed to inialize CLR host.");
+
+        try
+        {
+            // Get a CLR function that can load an assembly.
+            status = hostfxr_get_runtime_delegate(
+                _hostContextHandle,
+                hostfxr_delegate_type.load_assembly_and_get_function_pointer,
+                out load_assembly_and_get_function_pointer loadAssembly);
+            CheckStatus(status, "Failed to get CLR load-assembly function.");
```

Comment:
>CheckStatus [](http://example.com/codeflow?start=12&length=11)

If it is an extension method, then we would not need to assign result to a variable.